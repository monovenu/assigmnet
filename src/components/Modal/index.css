.modal{
    position:fixed;
    left:0;
    right:0;
    top:0;
    bottom:0;
    background-color:rgba(0,0,0,0.5);
    z-index:1050;
    opacity:0.2;
    transition: all .4s ease-out 0s;
}
.dialog{
    width:640px;
    min-height:300px;
    position:absolute;
    top: 50%;
    left: 50%;
    box-shadow:0 5px 15px rgba(0,0,0,.5);
    border-radius:2px;
    background-color:#fff;
    transform: translate(-50%,-40%);
    transition: all .4s ease-out 0s;
}
.dialog-header{
    box-sizing:border-box;
    font-size: 15px;
    font-weight: 700;
    border-bottom:1px solid #ddd;
    height: 50px;
    line-height: 50px;
    padding: 0 14px;
    .closeBtn{
        float:right;
        margin-right:10px;
        overflow: hidden;
        line-height:50px;
        cursor:pointer;
        color:#aaa;
    }
}

.in{
    opacity: 1;
}
.in .dialog{
    top: 50%;
    transform: translate(-50%, -50%);
}