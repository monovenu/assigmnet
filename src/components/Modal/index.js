import { useState, useEffect } from "react";
import './index.css';


const Modal = ({
        close,
        width=600,
        title='title',
        children
    }) => {
    const [showIn, setShowIn] = useState(false)

    useEffect(() => {
        show();
        return hide;
    }, [])

    const show = () => {
        setShowIn(true)
    }
    const hide = () => {
        setShowIn(false)
        const timer = setTimeout(function(){
            close()
            clearTimeout(timer);
        }, 200);
    }


    return <div className={`modal ${showIn ? 'in' : ''}`}>
    <div className="dialog" style={{width:width + 'px'}}>
        <header className="dialog-header">
            {title}
            <i className="iconfont icon-guanbi closeBtn"  onClick={hide} />
        </header>
        <div className="dialog-content">
            {React.cloneElement(children, {hide})}
        </div>
    </div>
</div>
}
export default Modal;

