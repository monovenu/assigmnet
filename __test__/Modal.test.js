import renderer from 'react-test-renderer';
import Modal from '../src/components/Modal';

it('Modal renders correctly', () => {
  const component = renderer.create(
    <Modal close={() => {}} title='Create SDK'>
      <div>test </div>
    </Modal>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});