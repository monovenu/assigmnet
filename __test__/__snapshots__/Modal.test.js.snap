// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders correctly 1`] = `
<div
  className="modal "
>
  <div
    className="dialog"
    style={
      {
        "width": "600px",
      }
    }
  >
    <header
      className="dialog-header"
    >
      Create SDK
      <i
        className="iconfont icon-guanbi closeBtn"
        onClick={[Function]}
      />
    </header>
    <div
      className="dialog-content"
    >
      <div
        hide={[Function]}
      >
        test 
      </div>
    </div>
  </div>
</div>
`;
